### ISSUE TITLE
problem using pymediainfo with Libreoffice

### ISSUE THREAD

**Issue by cwaynedavis1946** [sbraz/pymediainfo#81]
I know this isn't your problem but will greatly appreciate any pointers.  I have a python script that uses pymediainfo.  Works perfectly when called from Python 3.7.4 but fails when called from a LibreOffice 6.3.4.2 macro.  Looks like a ctype incompatibility.  LibreOffice 6.3.4.2 runs Python 3.5.9.  Error I get is:
```
  File "C:\Program Files\LibreOffice\program\python-core-3.5.9\lib\site-packages\pymediainfo\__init__.py", line 188, in _get_library
    lib = lib_type(library)
  File "C:\Program Files\LibreOffice\program\python-core-3.5.9\lib\ctypes\__init__.py", line 351, in __init__
    self._handle = _dlopen(self._name, mode)
  OSError: [WinError 193] %1 is not a valid Win32 application
```
Is there an earlier version compatible with Python 3.5.9 or is there a workaround?  I already have a version working with my python script running the MediaInfo.exe CLI as a subprocess but the CLI gives less accurate results than the MediaInfo GUI and pymediainfo versions.

I will greatly appreciate any help or suggestions.

Thank you,

Wayne

**Comment by sbraz** [comment]
Hi,
Looking at your error, I'd say LibreOffice's Python is 32-bit and your MediaInfo DLL is 64-bit. Can you check that?

Running pymediainfo from Python 3.5 isn't a problem, I even used to support Python 2.6 :) I don't use much fancy syntax that would require a newer version and I doubt ctypes changed a lot since 3.5.

**Comment by cwaynedavis1946** [comment]
Problem solved.  One was I had LibreOffice x64 and python x32.  That caused a DLL conflict.  Spotted by Louis Sautier, thanks.  After that created venv, installed pymediainfo in it.  Created venv\bin and copied python.exe to it.  Inserted:

#!path_to_venv\.getmivenv\bin\python.exe

into my python script.  Ran it from inside LibreOffice and it worked.  Didn't know python followed #!, also didn't know about venv before.  I'm learning python on the fly.

**Comment by JeromeMartinez** [comment]
>  MediaInfo.exe CLI as a subprocess but the CLI gives less accurate results than the MediaInfo GUI and pymediainfo versions.

Side note: MediaInfo CLI is just a CLI interface for MediaInfo library, at the same level as MediaInfo GUI or pymediainfo.
So if such difference exists (especially between MediaInfo GUI and MediaInfo CLI), please open [an issue ticket on MediaInfo GitHub](https://github.com/MediaArea/MediaInfoLib/issues).

**Comment by sbraz** [comment]
Maybe he was forgetting to pass it `-f` and `--output=OLDXML`.

**Comment by cwaynedavis1946** [comment]
No, ran CLI and GUI against same files.  I think I saw some inconsistencies  on formats and codecs.  Major problem was CLI didn't report audio and text languages that GUI did.  Tonight I'll try to come up with some cases and post one or more.  I just thought that no one had time to maintain the CLI and that GUI and pymediainfo might be on different branches.  Not complaining about any of this.  I'm just learning in this area.  From what I've found mediainfo is outstanding and is the best at what it does.  Thank you all for your work.
